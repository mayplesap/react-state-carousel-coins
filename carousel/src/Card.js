import React from "react";
import "./Card.css";

/** Creates a card with props of:
 * - caption - string
 * - src - image
 * - currNum - number 
 * - totalNum - number
 */
function Card(props) {
  return (
    <div className="Card">
      <h4 className="Card-title">{props.caption}</h4>
      <img className="Card-image" src={props.src} alt={props.caption} />
      <small className="Card-small">
        Image {props.currNum} of {props.totalNum}.
      </small>
    </div>
  );
}

export default Card;
