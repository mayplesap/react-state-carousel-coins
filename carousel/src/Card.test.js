import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

it("renders without crashing", function() {
  render(<Card />);
})

it("matches snapshot", function() {
  const { container } = render(<Card caption="test caption"
                                      src="image1"
                                      currNum="1"
                                      totalNum="1"/>);
  expect(container).toMatchSnapshot();
})

it("rendered card has correct info", function() {
  const { container } = render(<Card caption="test caption"
                                      src="image1"
                                      currNum="1"
                                      totalNum="1"/>);
  expect(container.querySelector("div")).toHaveClass("Card");
  expect(container.querySelector("h4")).toHaveTextContent("test caption");
})
