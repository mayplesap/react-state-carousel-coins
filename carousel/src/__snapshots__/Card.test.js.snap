// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test caption
    </h4>
    <img
      alt="test caption"
      class="Card-image"
      src="image1"
    />
    <small
      class="Card-small"
    >
      Image 
      1
       of 
      1
      .
    </small>
  </div>
</div>
`;
